test/jdk/jdk/crac/LazyProps.java

```diff
@@ -22,34 +22,39 @@
  */
 
 import jdk.crac.*;
-import jdk.test.lib.process.ProcessTools;
-import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.crac.CracBuilder;
+import jdk.test.lib.crac.CracEngine;
+import jdk.test.lib.crac.CracTest;
 
 /**
  * @test
  * @library /test/lib
  * @run main LazyProps
  */
-public class LazyProps {
-    static public void main(String[] args) throws Exception {
-        if (args.length == 0) {
-            OutputAnalyzer output = ProcessTools.executeTestJvm(
-                    "-XX:CREngine=simengine", "-XX:CRaCCheckpointTo=./cr",
-                    "LazyProps",
-                    "-runTest");
-            output.shouldHaveExitValue(0);
-            output.shouldContain("jdk.crac beforeCheckpoint");
-        } else {
-            Resource resource = new Resource() {
-                @Override
-                public void beforeCheckpoint(Context<? extends Resource> context) throws Exception { }
-                @Override
-                public void afterRestore(Context<? extends Resource> context) throws Exception { }
-            };
-            Core.getGlobalContext().register(resource);
+public class LazyProps implements CracTest {
+    public static void main(String[] args) throws Exception {
+        CracTest.run(LazyProps.class, args);
+    }
+
+    @Override
+    public void test() throws Exception {
+        new CracBuilder().engine(CracEngine.SIMULATE).main(LazyProps.class).args(CracTest.args())
+                .captureOutput(true)
```

Comment:
the same question, would it harm to make this default? Or is it false by default in the existing JDK testing libraries?

Reply:
Many tests don't use process output, and it was useful for debugging to have it printed straight out. If the process gets stuck I won't see that (right now we process the output only after it exits) but there's probably a way to both process it and print it out.

Reply:
OK, thanks, something like this perfectly describes why it should not be default. Being explicit is not bad.